### PR TITLE
Adding a local tiler.

### PR DESCRIPTION
**Pull request by lossyrob**

### CODE REVIEW

File: server/local-tiler.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    .then(function() {
+      // Keep the directory, we will clean it up manually.
+      var tmpOptions = { };//unsafeCleanup : true };
+      if (input.tmpDir) { 
+        tmpOptions.dir = input.tmpDir; 
+      }
+
+      return tmp.dirAsync(tmpOptions);
+    })
+    .then(function(tmpObj) {
```
Comment (sethfitz) [inline server/local-tiler.js]: [`spread`](https://github.com/petkaantonov/bluebird/blob/master/API.md#spreadfunction-fulfilledhandler--function-rejectedhandler----promise) might make this slightly more readable.

File: server/local-tiler.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                  return this.activity("translate", "1.0", inputPath, outputPath, { outputFormat: "PNG" });
+                }, {  concurrency: os.cpus().length })
+                .then(function(files) {
+                  this.log("Generated %d cell(s) for zoom %d at %s.", files.length, zoom, outputPath);
+
+                  return this.activity("buildVRT", "1.0", files, outputPath);
+                });
+            });
+        })
+        .finally(function() {
```
Comment (sethfitz) [inline server/local-tiler.js]: Try adding

``` javascript
.catch(function(err) {
  console.warn(err.stack);
  throw err;
})
```

to see if it spots anything we're missing.

**Comment by sethfitz** [comment]
I will bet you that the problem lies here: https://github.com/stamen/swfr/blob/783647a67cca74315112ce26a29344fadef987fb/lib/activities/resample.js#L65-L73

Specifically, that the dataset opened by `gdal.open()` is never [closed](http://naturalatlas.github.io/node-gdal/classes/gdal.Dataset.html#method-close).

**Comment by lossyrob** [comment]
@hectcastro merged in your changes. I changed the Dockerfile to not use the symlink on the `/npm/node_modules`, let me know if that's a bad change.

**Comment by lossyrob** [comment]
Merging this in now, preceding the repository breakup.

**Pull request merged**